coax/policy_objectives/_base.py

```diff
@@ -82,9 +82,15 @@ def loss_func(params, state, hyperparams, rng, transition_batch, Adv):
             if self.regularizer is not None:
                 hparams = hyperparams['regularizer']
                 W = jnp.clip(transition_batch.W, 0.1, 10.)  # clip imp. weights to reduce variance
-                loss = loss + jnp.mean(W * self.regularizer.function(dist_params, **hparams))
+                regularizer, regularizer_metrics = self.regularizer.batch_eval(params,
+                                                                               hparams,
+                                                                               state,
+                                                                               rng,
+                                                                               transition_batch)
+                loss = loss + jnp.mean(W * regularizer)
```

Comment:
It's a shame that we need to do another forward pass. Is this because we need to construct the `dist_params` for the intermediate steps?

I've been meaning to change the `TransitionBatch` and tracers to record `dist_params` instead of `logp` (we can always construct `logP` from `A` and `dist_params`), which would make thing like this more efficient.

I'll do that if I have time on the weekend.

Reply:
Yes, otherwise we would have to combine the entropy of the current state `s_{t}` with the entropy from the following states `s_{t:t+n}` in the regularizer. We can do that but that would make the implementation slightly more complicated. Recording the `dist_params` sounds like a good idea in either case!

Reply:
I just noticed that the `dist_params` would not help in the `TransitionBatch` because we need to compute the entropy on the current policy.

Reply:
Right of course.